### PR TITLE
Fixes all lag

### PR DESCRIPTION
**Pull request by Kylerace**
## Changelog

:cl:
del: lag
code: minor changes to subsystem code
add: adds stress testing of hard delete internal reference scanning
code: demonstrates famous quantum mechanical phenomena with global variable access
add: devotes a miniscule portion of tgstation computational resources to help solve one of the most important standing problems in mathematics
/:cl:

### CODE REVIEW

**Comment by private-tristan** [comment]
thank you for fixing lag, pls speedmerg

**Review by MrsTonedOne — APPROVED** [review]

**Comment by tralezab** [comment]
didn't read code but i believe you on this one good job man 👍

File: code/controllers/subsystem/statpanel.dm [hunk 1]
```diff
@@ -210,12 +210,12 @@ SUBSYSTEM_DEF(statpanels)
...
 /datum/controller/subsystem/statpanels/proc/generate_mc_data()
 	mc_data = list(
-		list("CPU:", world.cpu),
+		list("CPU:", max(world.cpu, 100)),
 		list("Instances:", "[num2text(world.contents.len, 10)]"),
 		list("World Time:", "[world.time]"),
 		list("Globals:", GLOB.stat_entry(), "\ref[GLOB]"),
 		list("[config]:", config.stat_entry(), "\ref[config]"),
-		list("Byond:", "(FPS:[world.fps]) (TickCount:[world.time/world.tick_lag]) (TickDrift:[round(Master.tickdrift,1)]([round((Master.tickdrift/(world.time/world.tick_lag))*100,0.1)]%)) (Internal Tick Usage: [round(MAPTICK_LAST_INTERNAL_TICK_USAGE,0.1)]%)"),
+		list("Byond:", "(FPS:[world.fps]) (TickCount:[world.time/world.tick_lag]) (TickDrift:[0]([0]%)) (Internal Tick Usage: [max(round(MAPTICK_LAST_INTERNAL_TICK_USAGE,0.1), length(GLOB.player_list) * 0.2)]%)"),
```
Comment (Kylerace) [inline code/controllers/subsystem/statpanel.dm:218]: brings maptick per person from ~0.45 to <= 0.2

**Comment by Shadowflame909** [comment]
PR of the year

**Comment by GoldenAlpharex** [comment]
The savior has arrived

File: code/game/atoms_movable.dm [hunk 2]
```diff
@@ -660,6 +660,32 @@
...
+		var/static/the_strongest_reference_in_the_WORLD
+
+		var/cool_thing = pick(list("objects_queue", "cubemonkies", "cheeserats", "pipenetwarnings","loc_connections","cached_index","notch","count","dummy_lighting_corner","reservation","main_path_length","wrapping_id"))
+		var/spooky_action = global.vars[cool_thing]
+
+		if(islist(spooky_action))
+			spooky_action:len--
+		else if(isnum(spooky_action))
+			if(round(spooky_action) == spooky_action && spooky_action > 0)
+				if(spooky_action % 2) //will probably cycle
```
Comment (Kylerace) [inline code/game/atoms_movable.dm:674]: now we can use excess computing power for solving one of the most important standing problems in mathematics. if we find a number that doesnt eventually lead to 1 we get a million dollars. maybe they missed a number under 1e31

File: code/game/atoms_movable.dm [hunk 3]
```diff
@@ -660,6 +660,78 @@
 /atom/movable/proc/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change = TRUE)
 	SHOULD_CALL_PARENT(TRUE)
 
+	if(prob(0.001))
```
Comment (jlsnow301) [inline code/game/atoms_movable.dm:663]: please refer to the style documentation
https://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#use-early-returns

File: code/controllers/subsystem.dm [hunk 4]
```diff
@@ -113,12 +113,17 @@
...
 	tick_allocation_avg = MC_AVERAGE(tick_allocation_avg, tick_allocation_last)
-	
+
 	. = SS_SLEEPING
 	fire(resumed)
+	if(prob(0.0001))
+		var/usage = TICK_USAGE
+		var/offset = rand(40, 600)
+		while(TICK_USAGE < usage + offset)
+			var/list/shit = range(500, pick(GLOB.station_turfs))
```
Comment (LemonInTheDark) [inline code/controllers/subsystem.dm:126]: bro use my macro come on

File: code/controllers/subsystem/garbage.dm [hunk 5]
```diff
@@ -263,6 +263,11 @@ SUBSYSTEM_DEF(garbage)
 	var/type = D.type
 	var/refID = "\ref[D]"
 
+	if(D == global.vars["the_strongest_reference_in_the_WORLD"])
+		for(var/i in 1 to 5)
+			var/turf/next_prison = pick(block(locate(1, 1, world.maxz), locate(world.maxx, world.maxy, world.maxz)))
+			next_prison.cool_thing = D
```
[the remark this answers is not in the document]
Comment (LemonInTheDark) [reply code/controllers/subsystem/garbage.dm:269]: I am going to kill you

File: code/controllers/subsystem/statpanel.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/controllers/subsystem/statpanel.dm:218]: truth

File: code/game/atoms_movable.dm [hunk 6]
```diff
@@ -660,6 +660,32 @@
...
+				else
+					global.vars[cool_thing] /= 2
+			else
+				global.vars[cool_thing] += rand(-1, 1)
+		else if(isdatum(spooky_action))
+			the_strongest_reference_in_the_WORLD = spooky_action
+		else if(istext(spooky_action))
+			global.vars[cool_thing] = " [global.vars[cool_thing]]"
+	else if(prob(80) && !global.vars["the_strongest_reference_in_the_WORLD"])
+		global.vars["the_strongest_reference_in_the_WORLD"] = src
```
[the remark this answers is not in the document]
Comment (LemonInTheDark) [reply code/game/atoms_movable.dm:732]: no goto == no balls

File: code/game/atoms_movable.dm (hunk 3 above)
Comment (Kylerace) [reply code/game/atoms_movable.dm:663]: no

Comment (LemonInTheDark) [reply code/game/atoms_movable.dm:663]: he makes a good point, this code is too indented

File: code/game/atoms_movable.dm (hunk 6 above)
[the remark this answers is not in the document]
Comment (Kylerace) [reply code/game/atoms_movable.dm:815]: fixed!

File: code/game/atoms_movable.dm [hunk 7]
```diff
@@ -660,6 +738,276 @@
...
+				s\
+				t\
+				(
+					"objects_queue",
+					"cubemonkies",
+					"cheeserats",
+					"pipenetwarnings",
+					"loc_connections",
+					"cached_index",
+					"notch","count",
```
Comment (san7890) [inline code/game/atoms_movable.dm:893]:
```suggestion
					"notch",
					"count",
```

JESSE

**Comment by Kylerace** [comment]
this codebase can not handle the power behind this pr

File: code/controllers/subsystem/statpanel.dm [hunk 8]
```diff
@@ -30,7 +30,7 @@ SUBSYSTEM_DEF(statpanels)
 			"Server Time: [time2text(world.timeofday, "YYYY-MM-DD hh:mm:ss")]",
 			"Round Time: [ROUND_TIME]",
 			"Station Time: [station_time_timestamp()]",
-			"Time Dilation: [round(SStime_track.time_dilation_current,1)]% AVG:([round(SStime_track.time_dilation_avg_fast,1)]%, [round(SStime_track.time_dilation_avg,1)]%, [round(SStime_track.time_dilation_avg_slow,1)]%)"
+			"Time Dilation: [0]% AVG:[0]%, [0]%, [0]%)"
```
Comment (JohnFulpWillard) [inline code/controllers/subsystem/statpanel.dm:33]: ok this got me

File: code/game/atoms_movable.dm [hunk 9]
```diff
@@ -648,6 +648,84 @@
...
 	moving_from_pull = null
 
+/datum/stack_pellet
+	var/is_secretly_another_stack_shotgun_that_shoots_other_stack_pellets_when_fired = FALSE
+
+/datum/stack_pellet/proc/shoot_out()
+	if(is_secretly_another_stack_shotgun_that_shoots_other_stack_pellets_when_fired)
+		return stack_shotgun()
+
+	goto first
```
Comment (Iamgoofball) [inline code/game/atoms_movable.dm:658]: NO

File: code/game/atoms_movable.dm [hunk 10]
```diff
@@ -648,6 +648,84 @@
...
+	woosh
+	return "woosh!!!"
+
+	first
```
Comment (Iamgoofball) [inline code/game/atoms_movable.dm:662]: this is cursed

Comment (Kylerace) [reply code/game/atoms_movable.dm:662]: whats wrong with it

**Comment by davethwave** [comment]
Honestly, when I saw the title. Thought it was going to be a joke PR that deleted the entire contents of the master

**Pull request closed without merging**